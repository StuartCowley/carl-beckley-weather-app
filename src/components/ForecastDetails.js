import React from "react";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();
  return (
    <div className="forecast-details">
      <div className="forecast-details_date">Date: {formattedDate}</div>
      <div className="forecast-details_temperature-max">
        Max Temp: {temperature.max}&deg;C
      </div>
      <div className="forecast-details_temperature-min">
        Min Temp: {temperature.min}&deg;C
      </div>
      <div className="forecast-details_humidity">Humidity: {humidity}</div>
      <div className="forecast-details_wind-speed">
        Wind Speed: {wind.speed}
      </div>
      <div className="forecast-details_wind-direction">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number.isRequired,
      direction: PropTypes.string.isRequired,
    }).isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number.isRequired,
      max: PropTypes.number.isRequired,
    }),
  }).isRequired,
};

export default ForecastDetails;
