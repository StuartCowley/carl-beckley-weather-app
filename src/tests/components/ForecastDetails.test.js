import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    date: 1111111,
    humidity: 30,
    temperature: {
      min: 10,
      max: 30,
    },
    wind: {
      speed: 20,
      direction: "s",
    },
  };

  it("should render the component", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
