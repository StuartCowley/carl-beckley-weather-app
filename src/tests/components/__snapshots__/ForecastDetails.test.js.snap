// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details should render the component 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details_date"
    >
      Date: Thu Jan 01 1970
    </div>
    <div
      class="forecast-details_temperature-max"
    >
      Max Temp: 30°C
    </div>
    <div
      class="forecast-details_temperature-min"
    >
      Min Temp: 10°C
    </div>
    <div
      class="forecast-details_humidity"
    >
      Humidity: 30
    </div>
    <div
      class="forecast-details_wind-speed"
    >
      Wind Speed: 20
    </div>
    <div
      class="forecast-details_wind-direction"
    >
      Wind Direction: s
    </div>
  </div>
</DocumentFragment>
`;
